src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java

```diff
@@ -459,4 +465,23 @@ private static String computeDifference(Set<?> set1, Set<?> set2) {
 		}
 		return StringUtils.join(results, "\n");
 	}
+	private void checkFields(CtModel model) {
```

Comment:
out of curiosity how long does this new check take?

Reply:
some testing shows around 60 seconds but I have an idea to speed it up.